src/main/java/cloud/fogbow/as/core/systemidp/DefaultSystemRolePlugin.java

```diff
@@ -0,0 +1,59 @@
+package cloud.fogbow.as.core.systemidp;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import cloud.fogbow.as.constants.ConfigurationPropertyKeys;
+import cloud.fogbow.as.constants.SystemConstants;
+import cloud.fogbow.as.core.PropertiesHolder;
+import cloud.fogbow.common.models.SystemUser;
+
+public class DefaultSystemRolePlugin implements SystemRolePlugin {
+
+	private Map<String, Set<String>> usersWithSpecialRoles;
+	private String defaultRole;
+	
+	public DefaultSystemRolePlugin() {
+		defaultRole = PropertiesHolder.getInstance().getProperty(ConfigurationPropertyKeys.DEFAULT_ROLE_KEY);
+		getUsersWithSpecialRoles();
+	}
+
+	private void getUsersWithSpecialRoles() {
+		usersWithSpecialRoles = new HashMap<String, Set<String>>();
+		String rolesNamesString = PropertiesHolder.getInstance().getProperty(ConfigurationPropertyKeys.ROLES_KEY);
+		
+		if (!rolesNamesString.isEmpty()) {
+			for (String roleName : rolesNamesString.trim().split(SystemConstants.ROLE_NAMES_SEPARATOR)) {
+				getUserNamesWithRole(roleName);
+			}			
+		}
+	}
+
+	private void getUserNamesWithRole(String roleName) {
+		String userNamesWithRoleString = PropertiesHolder.getInstance().getProperty(roleName);
+		
+		if (!userNamesWithRoleString.isEmpty()) {
+			for (String userName : userNamesWithRoleString.trim().split(SystemConstants.ROLE_NAMES_SEPARATOR)) {
+				if (!usersWithSpecialRoles.containsKey(userName)) {
+					usersWithSpecialRoles.put(userName, new HashSet<String>());
+				}
+				
+				usersWithSpecialRoles.get(userName).add(roleName);
+			}					
+		}
+	}
+	
+	public void setUserRoles(SystemUser user) {
+		String userName = user.getName();
+		
+		if (usersWithSpecialRoles.containsKey(userName)) {
+			user.setUserRoles(usersWithSpecialRoles.get(userName));			
```

Comment:
I am not following this logic, but it may well be my lack of deep knowledge on Java. I understand that usersWithSpecialRoles is a map role -> set of users with that role. Thus, I'd expect that you would need to iterate on all keys in the map (i.e. role names), and for each one you would check whether the user should be assigned the particular role (i.e. the set associated with the key contains the username). If so, you should add the role to the set of roles of the user. After iterating on the whole keys, you would add the set of roles to the user, by calling setUserRoles. Does the above code do this?

Reply:
usersWithSpecialRoles is a map user -> set of roles. I think role attribution is more efficient by using a data structure in this format, since it does not need to iterate on multiple keys.

Reply:
Ooops ... my bad. The names of the variables are quite clear, but I had my mind preset with what I thought would be the solution :-( It makes perfect sense. Please, disregard my comment.